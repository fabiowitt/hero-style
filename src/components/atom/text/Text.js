import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import css from './Text.module.scss';

const Text = ({ center, end, children }) => {
  const classes = classNames({
    [css['c-text']]: true,
    [css['c-text--center']]: center,
    [css['c-text--end']]: end
  });

  return <p className={classes}>{children}</p>;
};

Text.propTypes = {
  center: PropTypes.bool,
  children: PropTypes.string,
  end: PropTypes.bool
};

Text.defaultProps = {
  center: false,
  children: null,
  end: false
};

export default Text;
