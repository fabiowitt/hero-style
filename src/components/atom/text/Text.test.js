import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Text from './Text';

describe('Text', () => {
  it('should match with the snapshot', () => {
    const children = 'Some information';
    const wrapper = shallow(<Text>{children}</Text>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
