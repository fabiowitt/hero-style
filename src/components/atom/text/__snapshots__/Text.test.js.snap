// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match with the snapshot 1`] = `
<p
  className="c-text"
>
  Some information
</p>
`;
